src/app/store/ProductList/ProductList.reducer.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import {
+    APPEND_PAGE,
+    UPDATE_PRODUCT_LIST_ITEMS,
+    UPDATE_LOAD_STATUS
+} from 'Store/ProductList';
+
+const initialState = {
+    pages: {},
+    isLoading: true
+};
+
+const ProductListReducer = (state = initialState, action) => {
+    const {
+        items,
+        currentPage,
+        isLoading
+    } = action;
+
+    if (items) {
+        items.forEach(({ attributes, variants }, i) => {
+            attributes.forEach(({ attribute_code, attribute_value }) => {
+                items[i][attribute_code] = attribute_value;
+            });
+
+            if (variants) {
+                variants.forEach(({ product: { attributes } }, j) => {
+                    if (attributes) {
+                        attributes.forEach(({ attribute_code, attribute_value }) => {
+                            items[i].variants[j].product[attribute_code] = attribute_value;
+                        });
+                    }
+                });
+            }
+        });
+    }
+
```

Comment:
Destruct action.

Reply:
actually here it adds a bit of context (like it is type of action), while others are used with meaning that these are just some data passed in action and then destructed, you could think that this `type` is also part of these data.. I do agree that it is debatable, just wanted to let you know about reasoning of it